src/main/kotlin/org/rust/ide/annotator/fixes/AddMissingSupertraitImplFix.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.presentation.ImportingPsiRenderer
+import org.rust.ide.presentation.PsiRenderingOptions
+import org.rust.ide.presentation.renderTraitRef
+import org.rust.ide.presentation.renderTypeReference
+import org.rust.ide.refactoring.implementMembers.generateMissingTraitMembers
+import org.rust.ide.utils.GenericConstraints
+import org.rust.ide.utils.import.import
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.descendantsOfType
+import org.rust.lang.core.psi.ext.resolveToBoundTrait
+import org.rust.lang.core.resolve.ref.pathPsiSubst
+import org.rust.lang.core.types.*
+import org.rust.lang.core.types.infer.substitute
+
+/**
+ * Implements missing supertrait(s) for a type in an impl item.
+ *
+ * ```
+ * trait A {}
+ * trait B: A {}
+ *
+ * struct S;
+ *
+ * impl B/*caret*/ for S {}
+ *
+ * // fix adds
+ * impl A for S {}
+ * ```
+ */
+class AddMissingSupertraitImplFix(implItem: RsImplItem) : LocalQuickFixAndIntentionActionOnPsiElement(implItem) {
+    override fun getText(): String = "Implement missing supertrait(s)"
+    override fun getFamilyName(): String = text
+
+    override fun invoke(
+        project: Project,
+        file: PsiFile,
+        editor: Editor?,
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val impl = startElement as? RsImplItem ?: return
+        val traitRef = impl.traitRef ?: return
+        val trait = impl.implementedTrait ?: return
+
+        val typeRef = impl.typeReference ?: return
+        val type = typeRef.type
+        val implLookup = impl.implLookup
+
+        val traits = mutableListOf<Pair<BoundElement<RsTraitItem>, RsTraitRef>>()
+        val substitutions = mutableListOf(pathPsiSubst(traitRef.path, trait.element))
```

Comment:
I guess that such collecting substitutions for all traits fails in the case when one trait should be implemented multiple times with different type arguments. Well, sounds like a rare case anyway